src/rpm/headers/header.rs

```diff
@@ -987,6 +989,11 @@ mod test {
                     offset,
                     IndexData::StringTag(sha1.to_owned()),
                 ),
+                IndexEntry::new(
+                    IndexSignatureTag::RPMSIGTAG_SHA256,
+                    offset,
+                    IndexData::StringTag(sha256.to_owned()),
```

Comment:
If the length of this string is 0, 8, 12, 16, 24, 32, 40, 48 (I stopped there but you get the gist), RPM fails to parse

Reply:
It looks like this carries through other tags.  If you start tweaking the length of the `RSA` signature, it exhibits the same behavior.

With both the SHA1 and SHA2 digests in the header, an RSA tag with length 6 (binary) fails to parse while other lengths parse successfully but fail to validate.